Add an option to handle HTTP scheme

di72nn opened the pull request:
Useful when you want to save a link from an app which opens links in web browsers by default.
The option is disabled by default.

di72nn committed:
Add an option to handle HTTP scheme

di72nn commented:
If the method used in this PR is acceptable, it is also possible to add a work-around option to fix #19.

dimtion commented:
Thanks for the PR, I'll try to review within the next few days.

di72nn committed:
Don't use Preferences for "Handle HTTP scheme" option

dimtion commented:
So first of all thanks a lot for the PR, and sorry for the late response. 

I have a few remarks : 
- In `MainActivity.java`, you create a private propriety `isHandlingHttpScheme`, if I read correctly, you use it in order to know the old state of "HTTP Scheme handling" when saving the settings in `saveSettings()`. I don't really understand the reason why you need this propriety. Why not simply use the method `isHandlingHttpScheme()` to check the current state ?  
- Also, I would move this business logic https://github.com/dimtion/Shaarlier/pull/23/commits/508d96950439a415bcbc795de4019e38d32d3d6f#diff-cba241f74441999ac05d6e0bea6cd7a5R87 into the method `setHandleHttpScheme()`, in order to simplify the method `saveSettings()`, what do you think ?
- If you have some time, here is the french translation of `Show Shaarlier in the list of web browser` :  `Afficher Shaarlier dans la liste des navigateurs web`

dimtion commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -77,6 +77,14 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="5dp"
+            android:checked="true"
```
Since the option is disabled by default, wouldn't it be more logical to set this to `android:checked="false"` ?

di72nn committed:
Minor updates to "handle HTTP scheme" feature

di72nn commented:
> I don't really understand the reason why you need this propriety. Why not simply use the method isHandlingHttpScheme() to check the current state ?

I thought `PackageManager` should not be called on a whim (`saveSattings()` is invoked from `onPause()` so it may be called multiple times in an activity's lifetime), but I guess it doesn't really matter. Changed.

> move this business logic ... into the method `setHandleHttpScheme()`

Done.

> french translation

Added.

> `android:checked="false"`

Done.

dimtion commented:
I don't think the overhead introduced in `getPackageManager()` is high since the package manager is already loaded in the application context, it might need further check but it seems alright to call it multiple time this method.
